debezium-server/debezium-server-eventhubs/src/test/java/io/debezium/server/eventhubs/EventHubsIT.java

```diff
@@ -0,0 +1,109 @@
+import com.azure.messaging.eventhubs.models.PartitionEvent;
+
+import io.debezium.server.DebeziumServer;
+import io.debezium.server.TestDatabase;
+import io.debezium.server.events.ConnectorCompletedEvent;
+import io.debezium.server.events.ConnectorStartedEvent;
+import io.debezium.util.Testing;
+import io.quarkus.test.junit.QuarkusTest;
+
+/**
+ * Integration test that verifies basic reading from PostgreSQL database and
+ * writing to Azure Event Hubs.
+ *
+ * @author Abhishek Gupta
+ */
+@QuarkusTest
+public class EventHubsIT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventHubsChangeConsumer.class);
+
+    private static final int MESSAGE_COUNT = 4;
+    private static final String CONSUMER_GROUP = "$Default";
+
+    protected static TestDatabase db = null;
+    protected static EventHubProducerClient producer = null;
+    protected static EventHubConsumerClient consumer = null;
+
+    {
+        Testing.Files.delete(EventHubsTestConfigSource.OFFSET_STORE_PATH);
+        Testing.Files.createTestingFile(EventHubsTestConfigSource.OFFSET_STORE_PATH);
+    }
+
+    @AfterAll
+    static void stop() {
+        if (db != null) {
+            db.stop();
+        }
+        if (producer != null) {
+            producer.close();
+        }
+        if (consumer != null) {
+            consumer.close();
+        }
+    }
+
+    @Inject
+    DebeziumServer server;
+
+    void setupDependencies(@Observes ConnectorStartedEvent event) {
+        String finalConnectionString = String.format("%s;EntityPath=%s",
+                EventHubsTestConfigSource.getEventHubsConnectionString(), EventHubsTestConfigSource.getEventHubsName());
+
+        producer = new EventHubClientBuilder().connectionString(finalConnectionString).buildProducerClient();
+        db = new TestDatabase();
+        db.start();
+    }
+
+    void connectorCompleted(@Observes ConnectorCompletedEvent event) throws Exception {
+        if (!event.isSuccess()) {
+            throw (Exception) event.getError().get();
+        }
+    }
+
+    @Test
+    public void testEventHubs() throws Exception {
+        Testing.Print.enable();
+
+        String finalConnectionString = String.format("%s;EntityPath=%s",
+                EventHubsTestConfigSource.getEventHubsConnectionString(), EventHubsTestConfigSource.getEventHubsName());
+
+        consumer = new EventHubClientBuilder().connectionString(finalConnectionString).consumerGroup(CONSUMER_GROUP)
+                .buildConsumerClient();
+
+        final List<PartitionEvent> expected = new ArrayList<>();
+
+        Awaitility.await().atMost(Duration.ofSeconds(EventHubsTestConfigSource.waitForSeconds())).until(() -> {
+            IterableStream<PartitionEvent> events = consumer.receiveFromPartition("0", MESSAGE_COUNT,
+                    EventPosition.latest());
+
+            events.forEach(event -> expected.add(event));
+            return expected.size() >= MESSAGE_COUNT;
```

Comment:
Could we add at least some basic assertion on the events themselves? Like the expected keys for instance?

Reply:
done, I have updated the IT to include this

Reply:
Thx.